tls_dialer.go

```diff
@@ -0,0 +1,60 @@
+package zk
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"net"
+	"os"
+	"time"
+)
+
+func CreateTLSConfig(rootCAFile, certFile, keyFile string) (*tls.Config, error) {
+	rootCABytes, err := os.ReadFile(rootCAFile)
+	if err != nil {
+		return nil, err
+	}
+
+	rootCA := x509.NewCertPool()
+	ok := rootCA.AppendCertsFromPEM(rootCABytes)
+	if !ok {
+		return nil, err
+	}
+
+	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
+	if err != nil {
+		return nil, err
+	}
+
+	return &tls.Config{
+		Certificates: []tls.Certificate{cert},
+		RootCAs:      rootCA,
+	}, nil
+}
+
+func GetTLSDialer(servers []string, dialer *net.Dialer, tlsConfig *tls.Config) (Dialer, error) {
```

Comment:
is this function then used as the option to use as an option? The notion would normally be that the surrounding library would pass in this dialer themselves. If we wanted to support TLS Dial in the zk lib. I would want to have this outside the main zk package and in a supporting package